### DISCUSSION TITLE
What is the reason for creating easepick?

### DISCUSSION THREAD

**Discussion by cjayyy** [easepick/easepick#9] [Q&A] [answered]
Hi! 

I wonder if there is some document/post describing differences between `Litepicker` and `easepick`? What led you to abandon the former project and create a new one? :)

**Comment by wakirin** [comment, marked as answer]
Hi!

Thanks for the question. I thought about writing something about it, but I didn't know if anyone needed it or not.

There are several differences between `Litepicker` and `easepick`:

1.`easepick` is modular and you can include only the packages you need and get a smaller size.
This can be useful for people who want to reduce the file size.
For example, you need:
- single picker: you can use [@easepick/core](https://easepick.com/packages/core)
- single picker with locked days: you can use [@easepick/lock-plugin](https://easepick.com/packages/lock-plugin)

With `Litepicker` you only have one file that includes all the features.

2. Plugins for `easepick` are also created in TS and have their own declarations.
With `Litepicker` you need to declare parameters in the main library and this is wrong:
https://github.com/wakirin/Litepicker/blob/33bbcbb62b91a40006172aee7e419f558a4a4dcb/src/interfaces.ts#L74-L85

Also `easepick` has a PluginManager (currently undocumented). This allows you to manage plugins in an already created picker (add, remove, change options).

3. `Litepicker` is designed to support ancient browsers, and this eventually affect the file size.
I want to make `easypick` target more modern browsers. At the moment at least these versions are:
<img width="761" alt="Screen Shot 2022-03-26 at 13 06 33" src="https://user-images.githubusercontent.com/7056055/160228930-8de0f1ee-b9cb-43e0-9469-e4ac2ad1c3dd.png">

4. Some other things have been changed, such as:
- [webpack](https://webpack.js.org/) replaced with [rollup.js](https://rollupjs.org/guide/en/)
- CSS styles are no longer included in the JS file.
- Picker renders in [Shadow DOM](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM)

Actually when I start working on `easepick` it was supposed to be a new version of `Litepicker`, but in the end I decided to publish it as a new picker due to many breaking changes.
So for now I want to develop this project. `Litepicker` will be a read-only repository once the missing features in `easepick` are implemented as plugins (like dropdowns, multi-select, etc.)

**Reply by cjayyy** [reply]
Awesome! 
Thanks for such a detail response. :)
